src/main/java/bio/terra/service/resourcemanagement/google/GoogleProjectService.java

```diff
@@ -461,4 +465,16 @@ private ServiceUsage serviceUsage() throws IOException, GeneralSecurityException
         }
         return operation;
     }
+
+    @VisibleForTesting
+    static void ensureValidProjectId(final String projectId) {
+        Preconditions.checkNotNull(projectId, "Project Id must not be null");
+
+        Preconditions.checkArgument(
+            projectId.matches("^[a-z][a-z0-9-]{5,29}(?<!-)$"),
```

Comment:
```suggestion
            projectId.matches("^[a-z][a-z0-9-]{4,28}[a-z0-9]$"),
```

I think it's clearer to use this format instead of relying on negative lookbehinds.

Reply:
I guess I was hoping to make it match more with the description but I don't feel super strongly about it.  Happy to update it

Reply:
I often find that advanced features of regex are a double-edged sword. They provide a lot of power but the trade-off is often legibility and performance. As a result I tend to stick to the simpler constructs. For example, the negative lookbehind in this case will take more steps to complete than the equivalent without (admittedly, this is not a performance critical section).